File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        if (k > s.length()) return false;
        unordered_set<string>l;
        int i = 0, j = k, m = s.length() + 1;
        while (j != m) {
            l.insert(s.substr(i, j - i));
            j += 1;
            i += 1;
        }
        if (l.size() == pow(2, k)) return true;
        return false;
        }
};

File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        if k > len(s): return False
        l = []
        i, j, m = 0, k, len(s)+1
        while j != m:
            l.append(s[i:j])
            j += 1
            i += 1
        # print(l)
        if len(set(l)) == 2**k : return True
        return False